### PR TITLE
Fix hostname checks in clickhouse-keeper

### LINKED ISSUE #35707: Cannot mix loopback and non-local hostnames in raft_configuration (ClickHouse-Keeper)
Hi!

I heard on the last release meetup that ClickHouse-Keeper was ready for production and I wanted to try it out. However, I am running into some issues that I cannot really explain.

My staging setup is 3 VMs running Ubuntu 20.04 in Vagrant (Virtualbox). I am deploying ClickHouse-Keeper into the VMs as docker containers running the image `clickhouse/clickhouse-server:22.3.2.2`. The configuration looks like this.

```xml
<clickhouse>
    <listen_host>0.0.0.0</listen_host>
    <keeper_server>
        <tcp_port>2181</tcp_port>
        <server_id>1</server_id>

        <log_storage_path>/var/lib/clickhouse/coordination/log</log_storage_path>
        <snapshot_storage_path>/var/lib/clickhouse/coordination/snapshots</snapshot_storage_path>

        <coordination_settings>
            <raft_logs_level>trace</raft_logs_level>
        </coordination_settings>

        <raft_configuration>
            <server>
                <id>1</id>
                <hostname>dev-zoo01</hostname>
                <port>9234</port>
            </server>
            <server>
                <id>2</id>
                <hostname>dev-zoo02</hostname>
                <port>9234</port>
            </server>
            <server>
                <id>3</id>
                <hostname>dev-zoo03</hostname>
                <port>9234</port>
            </server>
        </raft_configuration>
    </keeper_server>
</clickhouse>
```

The docker container is running with `--network=host` on each of the 3 Ubuntu VMs.

This is the same setup that I had when running ClickHouse-Keeper in docker containers locally which worked fine without issue. 
However,  in the VMs I am running into this error immediately on startup. Additionally, I ran into the same error when testing on production VMs (same setup which worked with Apache Zookeeper).

```
Processing configuration file '/etc/clickhouse-keeper/config.xml'.
Sending crash reports is disabled
Starting ClickHouse Keeper 22.3.2.1 with revision 54460, build id: B537AFA18EAC3AF4, PID 1
starting up
OS Name = Linux, OS Version = 5.4.0-65-generic, OS Architecture = x86_64
DB::Exception: Mixing loopback and non-local hostnames ('dev-zoo01' and 'dev-zoo03') in raft_configuration is not allowed. Different hosts can resolve it to themselves so it's not allowed.
shutting down
Stop SignalListener thread
```

This error is very strange to me and not something that I have encountered before. It seems to be caused by the following line the `/etc/hosts` on the ubuntu VMs

```
127.0.1.1 dev-zoo01.company.com dev-zoo01
```

If I remove that line from the `/etc/hosts` file it seems to work but I am not sure how that will affect the rest of the system. Is this expected behavior? What is the recommended solution to this?

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Bug Fix (user-visible misbehaviour in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix hostname sanity checks for Keeper cluster configuration.
Add `keeper_server.host_checks_enabled` config to enable/disable those checks.

Fix #35707 

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

**Review by alesapin — APPROVED** [review]

File: src/Coordination/KeeperStateManager.cpp [hunk 1]
```diff
@@ -47,18 +33,19 @@ bool isLocalhost(const std::string & hostname)
...
-        "interserver_https_port",
-        "tcp_port",
-        "tcp_with_proxy_port",
-        "tcp_port_secure",
-        "mysql_port",
-        "postgresql_port",
-        "grpc_port",
-        "prometheus.port",
+    using namespace std::string_literals;
+    static const std::array config_port_names = {
```
Comment (alesapin) [inline src/Coordination/KeeperStateManager.cpp:37]: each time I read your PR I learn something new about C++.

Comment (antonio2368) [reply src/Coordination/KeeperStateManager.cpp:37]: Glad to hear that 😄

**Comment by alesapin** [comment]
@mergifyio update

**Comment by alesapin** [comment]
Errors unrelated to changes.

**Pull request merged**